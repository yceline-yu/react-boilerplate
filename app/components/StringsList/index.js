import React from 'react';
import PropTypes from 'prop-types';

import List from 'components/List';
import ListItem from 'components/ListItem';
import LoadingIndicator from 'components/LoadingIndicator';
import StringListItem from 'containers/StringListItem';

function StringsList({ loading, error, strings }) {
  if (loading) {
    return <List component={LoadingIndicator} />;
  }

  if (error !== false) {
    const ErrorComponent = () => (
      <ListItem item="Something went wrong, please try again!" />
    );
    return <List component={ErrorComponent} />;
  }

  if (strings !== false) {
    if (strings.strings.length === 0) {
      const EmptyComponent = () => <ListItem item="The list is empty" />;
      return <List component={EmptyComponent} />;
    }
    return <List items={strings.strings} component={StringListItem} />;
  }

  return null;
}

StringsList.propTypes = {
  loading: PropTypes.bool,
  error: PropTypes.any,
  strings: PropTypes.any,
};

export default StringsList;
