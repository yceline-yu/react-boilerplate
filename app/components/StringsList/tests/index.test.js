import React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { render } from 'react-testing-library';

import StringsList from '../index';
import configureStore from '../../../configureStore';

describe('<StringsList />', () => {
  it('should render the loading indicator when its loading', () => {
    const { container } = render(<StringsList loading />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render an error if loading failed', () => {
    const { queryByText } = render(
      <IntlProvider locale="en">
        <StringsList loading={false} error={{ message: 'Loading failed!' }} />
      </IntlProvider>,
    );
    expect(queryByText(/Something went wrong/)).not.toBeNull();
  });

  it('should render the strings if loading was successful', () => {
    const store = configureStore({}, browserHistory);
    const strings = { strings: ['test', 'test'] };
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <StringsList strings={strings} error={false} />
        </IntlProvider>
      </Provider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not render anything if nothing interesting is provided', () => {
    const { container } = render(
      <StringsList strings={false} error={false} loading={false} />,
    );

    expect(container.firstChild).toBeNull();
  });
});
