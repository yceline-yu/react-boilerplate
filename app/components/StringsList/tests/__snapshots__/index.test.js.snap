// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StringsList /> should render the loading indicator when its loading 1`] = `
<div
  class="Wrapper-sc-1umgotm-0 jjVzQg"
>
  <ul
    class="Ul-mwnq6h-0 jcxXFy"
  >
    <div
      class="Wrapper-sc-12uw37d-0 bTWIVF"
    >
      <div
        class="Circle__CirclePrimitive-ww56dy-0 enQIMS"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 jCxmwS"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 dyFXB"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 ikwUeP"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 hwlQRz"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 jjXxyb"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 jVExkJ"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 haCVIz"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 ghhiLZ"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 bCbqdt"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 hZPQMA"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 fsGOmP"
      />
    </div>
  </ul>
</div>
`;

exports[`<StringsList /> should render the repositories if loading was successful 1`] = `
<div
  class="Wrapper-sc-1umgotm-0 jjVzQg"
>
  <ul
    class="Ul-mwnq6h-0 jcxXFy"
  >
    <li
      class="Wrapper-euo0oy-0 gpjGxP"
    >
      <div
        class="Item-sc-3y9mie-0 jESnYC"
      >
        <div
          class="Wrapper-sc-4ooklm-0 eeXvdc"
        >
           
          test
           
        </div>
      </div>
    </li>
    <li
      class="Wrapper-euo0oy-0 gpjGxP"
    >
      <div
        class="Item-sc-3y9mie-0 jESnYC"
      >
        <div
          class="Wrapper-sc-4ooklm-0 eeXvdc"
        >
           
          test
           
        </div>
      </div>
    </li>
  </ul>
</div>
`;
